src/Storages/MergeTree/MergeTreeIndexSet.cpp

```diff
@@ -326,6 +332,26 @@ bool MergeTreeIndexConditionSet::mayBeTrueOnGranule(MergeTreeIndexGranulePtr idx
 }
 
 
+static const ActionsDAG::NodeRawConstPtrs & getArguments(const ActionsDAG::Node & node, const ActionsDAGPtr & result_dag_or_null, ActionsDAG::NodeRawConstPtrs * storage)
+{
+    chassert(node.type == ActionsDAG::ActionType::FUNCTION);
+    if (node.function_base->getName() != "indexHint")
+        return node.children;
+
+    /// indexHint arguments are stored inside of `FunctionIndexHint` class.
+    const auto & adaptor = typeid_cast<const FunctionToFunctionBaseAdaptor &>(*node.function_base);
+    const auto & index_hint = typeid_cast<const FunctionIndexHint &>(*adaptor.getFunction());
+    if (!result_dag_or_null)
+        return index_hint.getActions()->getOutputs();
+
+    /// Import the DAG and map argument pointers.
+    ActionsDAGPtr actions_clone = index_hint.getActions()->clone();
+    chassert(storage);
+    *storage = actions_clone->getOutputs();
+    result_dag_or_null->mergeNodes(std::move(*actions_clone));
```

Comment:
Will the node pointers copied to `store` remain valid after actions are passed to `mergeNodes`?